src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -394,79 +376,50 @@ std::vector<VectorDims> Snippet::shapeInfer() const {
         IE_THROW() << errorMessage.str();
     }
 
+    std::vector<ov::Shape> new_shapes;
+    for (const auto& s : normInputShapes)
+        new_shapes.emplace_back(s);
+    copy_snippet();
+    const auto& outputShapes = snippetAttrs.snippet->reshape_body(new_shapes);
```

Comment:
If I'm not missing anything, `ShapeInfer()` would be called every time `inputShapesModified()`. So currently it's every time input shapes are updated, even when we have them in the cache and don't actually need to perform expensive operations such as `copy_snippet()` and `reshape_body()`. Please correct me if I'm mistaken.

My second concern is, even when we really need to make shape infer (new set of shapes not in the cache), why do we need to `copy_snippet` every time? Why can't we reshape the original body?

Reply:
As the local body is modified through out shapeInfer(), canonicalization() etc based on previous input shapes. This lead for example some broadcast or other snippets dialect is insert. At current shapeInfer(), canonicalization() can not based on the modified body. seems like can not reshape back to original topology. I originally always use the single local body, there are lot of failures in test, so restart from original for new shape. Thank you!